server/neptune/gateway/event/check_run_handler.go

```diff
@@ -102,27 +129,66 @@ func (h *CheckRunHandler) signalPlanReviewWorkflowChannel(ctx context.Context, e
 	return nil
 }
 
-func (h *CheckRunHandler) signalUnlockWorkflowChannel(ctx context.Context, event CheckRun) error {
-	// Parse out root from check run name
-	matches := checkRunRegex.FindStringSubmatch(event.Name)
-	if len(matches) != 2 {
-		return fmt.Errorf("unable to determine root name")
-	}
-	rootName := matches[checkRunRegex.SubexpIndex("name")]
-	ctx = context.WithValue(ctx, contextInternal.ProjectKey, rootName)
-	err := h.TemporalClient.SignalWorkflow(
+func (h *CheckRunHandler) signalUnlockWorkflowChannel(ctx context.Context, event CheckRun, rootName string) error {
+	err := h.DeploySignaler.SignalWorkflow(
 		ctx,
 		// deploy workflow id is repo||root (the name of the check run is the root)
 		buildDeployWorkflowID(event.Repo.FullName, rootName),
 		// keeping this empty is fine since temporal will find the currently running workflow
 		"",
 		workflows.DeployUnlockSignalName,
 		workflows.DeployUnlockSignalRequest{
-			User: event.User,
+			User: event.User.Username,
 		})
 	if err != nil {
 		return errors.Wrapf(err, "signaling workflow with id: %s", event.ExternalID)
 	}
 	h.Logger.InfoContext(ctx, fmt.Sprintf("Signaled workflow with id %s to unlock", event.ExternalID))
 	return nil
 }
+
+func (h *CheckRunHandler) buildRoot(ctx context.Context, event CheckRun, rootName string) error {
+	builderOptions := BuilderOptions{
+		RepoFetcherOptions: github.RepoFetcherOptions{
+			ShallowClone: true,
+		},
+		FileFetcherOptions: github.FileFetcherOptions{
+			Sha: event.HeadSha,
+		},
+	}
+	rootCfgs, err := h.RootConfigBuilder.Build(ctx, event.Repo, event.Branch, event.HeadSha, event.InstallationToken, builderOptions)
+	if err != nil {
+		return errors.Wrap(err, "generating roots")
+	}
+	for _, rootCfg := range rootCfgs {
+		if rootCfg.Name != rootName {
+			continue
+		}
+		c := context.WithValue(ctx, contextInternal.ProjectKey, rootCfg.Name)
+		if rootCfg.WorkflowMode != valid.PlatformWorkflowMode {
+			h.Logger.DebugContext(c, "root is not configured for platform mode, skipping...")
+			continue
+		}
+		deployOptions := RootDeployOptions{
+			Repo:              event.Repo,
+			Branch:            event.Branch,
+			Revision:          event.HeadSha,
+			Sender:            event.User,
+			InstallationToken: event.InstallationToken,
+			BuilderOptions:    builderOptions,
+			Trigger:           workflows.MergeTrigger,
```

Comment:
True, but I saw it as re-running what originally was a merged revision, so the only difference between the previous revision request and the new one is that we set `Rerun` field to true. We can change this, but it it'll add complexity within the workflow's code since we currently equate `ManualTrigger` to Force Applies in all of the logic decisions there.

Reply:
The more I think about it, I wonder if it should be equated to force applies.  If you think about it in terms of how we do code deploys (ie. retrying a deploy) it's just another manual deploy.   Locking makes sense, if you want to keep retrying because of some random transient error and someone else merges into the queue.

We can leave the re-run information in there though still cuz I think we should still be safe about only allowing identical revisions.


Also last thing to note is that we're going to overwrite check runs effectively here which is probably fine initially, I'm not super sure how we would solve that and I don't want to have to keep track of a retry count or anything to use for the status name.

Reply:
sounds good, checkrun/suite should now be manual types.